src/main/java/com/google/sps/servlets/PhotosServlet.java

```diff
@@ -73,22 +57,13 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       maxPhotos = Integer.MAX_VALUE;
     }
 
-    // Make sure maxPhotos is not negative.
-    if (maxPhotos < 0) {
-      maxPhotos = 0;
-    }
-
-    // Load current user's photos from datastore.
-    String userId = loggedUser.getId();
-    Query query = new Query("BlurImage");
-    query.setFilter(new Query.FilterPredicate("userId", Query.FilterOperator.EQUAL, userId));
-    query.addSort("dateCreated", SortDirection.DESCENDING);
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
+    // Load current user's photos from datastore ordered by date created.
+    Iterable<Entity> photoEntities = loggedUser.getBlurImageEntities(maxPhotos);
 
     // Store photos in an array.
     ArrayList<BlurImage> photos = new ArrayList<>();
-    for (Entity entity : results.asIterable(FetchOptions.Builder.withLimit(maxPhotos))) {
+    for (Entity entity : photoEntities) {
+      String userId = loggedUser.getId();
```

Comment:
Please move it out of the loop.

Reply:
I'm not sure why I even put that inside the loop.

Solved, thanks!